openfisca_core/formulas.py

```diff
@@ -774,15 +766,11 @@ def new_filled_column(
     elif isinstance(source_file_path, str):
         source_file_path = source_file_path.decode('utf-8')
 
-    if start_date is UnboundLocalError:
-        start_date = None if reference_column is None else reference_column.start
-    elif start_date is not None:
-        assert isinstance(start_date, datetime.date)
-
-    if stop_date is UnboundLocalError:
-        stop_date = None if reference_column is None else reference_column.end
-    elif stop_date is not None:
-        assert isinstance(stop_date, datetime.date)
+    if end is UnboundLocalError:
+        end = None if reference_column is None else reference_column.end
+    elif end is not None:
+        assert isinstance(end, str)
+        get_datetime_date(name, end)  # Raises ValueError on format error.
```

Comment:
To avoid adding dependencies between `new_filled_column` sections it seems better to avoid declaring an end_date there. 
And, it is better to extract this code in a helper because the same instruction has to be called 2 times for 2 different things (checking the attribute format and giving a date time date from a string).